networkit/cpp/embedding/LearnEmbedding.cpp

```diff
@@ -0,0 +1,221 @@
+/*
+ * LearnEmbeddings.cpp
+ *
+ *  Created on: 06.07.2020
+ *      Author: Klaus Ahrens  <ahrens@informatik.hu-berlin.de>
+ *
+ *  adapted and reimplemented from node2vec
+ *  part of snap [https://github.com/snap-stanford/snap]
+ *  Copyright (c) 2007-2019, Jure Leskovec (under BSD license)
+ *
+ *  see [https://arxiv.org/pdf/1607.00653v1.pdf]
+ *
+ */
+
+// networkit-format
+
+#include <algorithm>
+#include <fstream>
+#include <unordered_map>
+#include <vector>
+
+#include <networkit/auxiliary/Random.hpp>
+
+#include "AliasSampler.hpp"
+#include "BiasedRandomWalk.hpp"
+#include "LearnEmbedding.hpp"
+
+// Code from https://github.com/nicholas-leonard/word2vec/blob/master/word2vec.c
+// Customized for SNAP and node2vec
+
+namespace NetworKit {
+namespace Embedding {
+
+// Number of negative samples. Value taken from original word2vec code.
+constexpr int negSamN = 5;
+
+constexpr int maxExp = 6;
+
+// Learning parameters for SGD. Values taken from original word2vec code.
+constexpr double startAlpha = 0.025;
+constexpr int refreshAlphaCount = 10000;
+constexpr double minAlpha = startAlpha * 0.0001;
+constexpr double Pwr = 0.75;
```

Comment:
Use camelCase for variables.

Reply:
Also, maybe explain how this variable is used/affects the algorithm, the name does not say much about it.